log/log.go

```diff
@@ -0,0 +1,96 @@
+package log
+
+import (
+	"io"
+	"os"
+
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+// Logger is the interface exposed to users of this launcher logger package.
+type Logger interface {
+	kitlog.Logger
+	// Debug logs keyvals at the "debug" level
+	Debug(keyvals ...interface{}) error
+	// Info logs keyvals at the "info" level
+	Info(keyvals ...interface{}) error
+	// Fatal logs keyvals at the "info" level, and then exits with error code 1.
+	Fatal(keyvals ...interface{}) error
+
+	// AllowDebug sets the allowed log level to include debug logs and above (info).
+	AllowDebug()
+	// AllowInfo sets the allowed log level to include info logs.
+	AllowInfo()
+}
+
+type logger struct {
```

Comment:
Do you think it would be possible to scale this package back in this way?

- Export the `logger` struct. 
- Return the struct. 
- Don't declare an interface in this package. 
- Only make any necessary changes in main, leaving other packages to only consume a go-kit logger? 

The reason I'm asking is because the logger is now required by all the packages in this repo, including ones meant to be imported by server-side implementations. 

 I wouldn't be opposed to discussing/iterating on the proposed interface here, but I think it should be more carefully considered before it propagates.

Reply:
For example, for GCP servers I have created this helper https://github.com/kolide/kit/blob/566c8f56a6ff7daba204818fbab0f2cb854b3310/logutil/logutil.go#L22 to rewrite the `level` key. I'm not sure how that will work with this logger.